cdm/src/main/java/ucar/nc2/ft2/coverage/simpgeometry/SimpleGeometryReader.java

```diff
@@ -27,9 +27,15 @@ public Polygon readPolygon(String name, int index) {
 		Variable polyvar = ds.findVariable(name);
 		if(polyvar == null) return null;
 		
-		// create a blank cf Polygon
-		Polygon poly = new CFPolygon();
-		return poly.setupPolygon(ds, polyvar, index);
+		Polygon poly = null;
+		
+		// CFConvention
+		if(ds.findGlobalAttribute(CF.CONVENTIONS) != null)
+			if(ucar.nc2.dataset.conv.CF1Convention.getVersion(ds.findGlobalAttribute(CF.CONVENTIONS).getStringValue()) >= 0)
+				poly = new CFPolygon();
+		
+		if(poly == null) return null;
+		else return poly.setupPolygon(ds, polyvar, index);

```

Comment:
What small changes higher up in the method would enable you to replace line 37 and 38 with a simple `return poly;`?